### ISSUE TITLE
New gcc won't compile libxml++ with -Werror 

### ISSUE THREAD

**Issue by elazarl** [nvmecompliance/tnvme#37]
When compiling with Werror, errors are thrown for using `autoptr`

For example
```
/usr/include/libxml++-2.6/libxml++/validators/relaxngvalidator.h:159:8: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
   std::auto_ptr<Impl> pimpl_;
        ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from Singletons/objRsrc.h:21,
                 from Singletons/rsrcMngr.h:21,
                 from globals.h:22,
                 from testResults.h:20,
                 from test.h:23,
                 from group.h:23,
                 from tnvmeParsers.h:20,
                 from tnvmeParsers.cpp:24:
```

**Comment by JeffHensel** [comment]
auto_ptr was depreciated (ie you would have to revert to older versions of gcc/boost/libxml) This repo should be updated to use new repos ie replace auto_ptr with unique_ptr

**Comment by elazarl** [comment]
@jhenselUNH would you accept a PR  that does that?

**Comment by JeffHensel** [comment]
PR? I believe the older version of ubuntu would have these libs pre-installed but I may be incorrect (12.x, 10.x)

EDIT: I think I am incorrect actually (I know at least gcc doesn't come pre-installed on those distros)

**Comment by elazarl** [comment]
The problem is not with compiling the package, per se, it would just be
nicer if this package would support newer compilers out of the box.

I want to submit a pull-request modernizing the code.

Would you be interested in merging such a pull request?

**Comment by JeffHensel** [comment]
I agree if you submit a change request it would be nice to have this repo updated to utilize the newer compiler.

This being said it will be up to the active members of the UNH-IOL to accept/decline the change (since they build on top of these repos we don't want to break any of their code).

Once you post it I can reach out to them about this
